### DISCUSSION TITLE
Only homepage is indexed

### DISCUSSION THREAD

**Discussion by Monsignor** [spatie/laravel-sitemap#375] [Q&A]
I ran the sitemap generator on my Laravel 7 app and, for some reason, it only indexed the homepage.

The links in the navigation are not generated by Javascript and they are not nofollow. I am using the default config for the generator, I didn't change a thing.

Yet, the generated sitemap.xml looks like this:

```
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
<url>
<loc>http://example.com/</loc>
<lastmod>2021-03-25T06:40:01+00:00</lastmod>
<changefreq>daily</changefreq>
<priority>0.8</priority>
</url>
</urlset>
```

**Comment by Luukdewaaier** [comment]
I am having te same issue, did you find a solution to this?

**Reply by Monsignor** [reply]
No, I had to create a manual solution for my site.

**Reply by Luukdewaaier** [reply]
Too bad, mind sharing the solution?

**Reply by Monsignor** [reply]
I could, but  I literally just added the pages of my website to the XML, I don't think this is useful to anyone :)

**Reply by Luukdewaaier** [reply]
Aah, that isn't useful indeed ;)

**Comment by pbgswd** [comment]
It wont add a collection. 

`        SitemapGenerator::create(config('app.url'))
            ->add(Post::all())
            ->writeToFile(public_path().'/sitemap.xml');`

Error
Call to undefined method Spatie\Sitemap\SitemapGenerator::add() 

 `
        Sitemap::create(config('app.url'))
            ->add(Post::all())
            ->writeToFile(public_path().'/sitemap.xml');`
BadMethodCallException
Method Illuminate\Database\Eloquent\Collection::getType does not exist. (View: /var/www/project118/laravel/vendor/spatie/laravel-sitemap/resources/views/sitemap.blade.php) 

I dont know why it would look for a blade template in the second example. All I want to do is add a collection to the sitemap and it does not work. Dont know why. 

Like others, I manually added links and called it good enough for now.
